.ci/Jenkinsfile

```diff
@@ -100,20 +104,69 @@ pipeline {
                 unsuccessful {
                   archiveArtifacts(allowEmptyArchive: true, artifacts: "${MSBUILDDEBUGPATH}/**/MSBuild_*.failure.txt")
                 }
+                always {
+                  archiveArtifacts(allowEmptyArchive: true, artifacts: "${BASE_DIR}/**/build/output/**/*.nupkg")
+                }
               }
             }
-            stage('Deploy') {
-              when {
-                branch 'master'
-              }
-              steps {
-                dir("${BASE_DIR}"){
-                  release('secret/apm-team/ci/elastic-observability-appveyor')
-                }
+          }
+        }
+      }
+    }
+    stage('Release') {
+      when {
+        beforeAgent true
+        anyOf {
+          tag pattern: '\\d+\\.\\d+\\.\\d+(-(alpha|beta|rc)\\d*)?', comparator: 'REGEXP'
+          branch 'master'
+        }
+      }
+      stages {
+        stage('Build') {
+          steps {
+            deleteDir()
+            unstash 'source'
+            dir("${BASE_DIR}"){
+              dotnet() {
+                sh(label: 'Generate local nuget packages', script: './build.sh generatepackages -s true')
+                sh(label: 'Validate *.npkg files that were created', script: './build.sh validatepackages -s true')
+                sh(label: 'Inspect public API change', script: './build.sh generateapichanges -s true')
               }
-              post {
-                always {
-                  archiveArtifacts(allowEmptyArchive: true, artifacts: "${BASE_DIR}/**/build/output/**/*.nupkg")
+            }
+          }
+        }
+        stage('Release to appveyor') {
+          when { branch 'master' }
+          steps {
+            dir("${BASE_DIR}"){
+              deploy(secret: 'secret/apm-team/ci/elastic-observability-appveyor')
+            }
+          }
+        }
+        stage('Release to feedz.io') {
+          when { branch 'master' }
+          steps {
+            dir("${BASE_DIR}"){
+              deploy(secret: 'secret/apm-team/ci/elastic-observability-feedz.io')
+            }
+          }
+        }
+        stage('Release to NuGet') {
+          when { branch 'master' }
+          steps {
+            dir("${BASE_DIR}"){
+              deploy('secret/apm-team/ci/elastic-observability-nuget')
+            }
+          }
+        }
+        stage('Release Notes') {
+          when { buildingTag() }
+          steps {
+            dir("${BASE_DIR}"){
+              withGitRelease(credentialsId: '2a9602aa-ab9f-4e52-baf3-b71ca88469c7-UserAndToken') {
+                dotnet() {
+                  sh(label: 'Generate release notes', script: './build.sh generatereleasenotes -s true')
```

Comment:
Is the output of `sh(label: 'Inspect public API change', script: './build.sh generateapichanges -s true')` available here as well?

Reply:
I copied from https://github.com/elastic/ecs-dotnet/pull/80/commits/6cc0494a914e01fe25bfc00589fd12967c1357e5#diff-e9f950f17198d3d5e3122a44230a09b9L75-L80 so , I'll include it then :)